src/Microsoft.Health.Fhir.Core/Features/Search/Converters/FhirElementToSearchValueTypeConverter.cs

```diff
@@ -38,14 +39,15 @@ public IEnumerable<ISearchValue> ConvertTo(object value)
 
             EnsureArg.IsOfType(value, typeof(TFhirElement), nameof(value));
 
-            return ConvertTo((TFhirElement)value);
+            return ConvertTo((TFhirElement)value, searchParameterType);
         }
 
         /// <summary>
         /// Converts the FHIR element to a list of <see cref="ISearchValue"/>.
         /// </summary>
         /// <param name="value">The value to convert.</param>
+        /// <param name="searchParameterType">The target type</param>
         /// <returns>A list of <see cref="ISearchValue"/>.</returns>
-        protected abstract IEnumerable<ISearchValue> ConvertTo(TFhirElement value);
+        protected abstract IEnumerable<ISearchValue> ConvertTo(TFhirElement value, SearchParamType searchParameterType);
```

Comment:
I was thinking about this. Instead of the ConvertTo method takes the serach parameter type, we could make two separate converters (e.g., `RangeToNumberSearchValueTypeConverter` and `RangeToQuantitySearchValueTypeConverter`) and have the `FhirElementToSearchValueTypeConverterManager` register the converters based on the ElementType + SearchParamType. That way each converter only does one thing and there won't be oddness with SearchParamType being passed in but ignored by most of the converters?